### PR TITLE
add nonzero narrow copy test

### PR DESCRIPTION
**Pull request by Krovatkin**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #80229
* __->__ #80228

narrow_copy -> narrow_copy_symint

### CODE REVIEW

File: test/cpp/lazy/test_lazy_ops.cpp [hunk 1]
```diff
@@ -7311,6 +7311,26 @@ TEST_F(LazyOpsTest, TestNarrowCopy) {
...
+  auto z_cpu = b_cpu.narrow_copy(0, 0, y0_cpu_size);
+
+  // Same operations, but on LAZY tensor
+  auto x_lazy = x_cpu.to(kLazy);
+  auto y_lazy = torch::nonzero(x_lazy);
+  // y_lazy is a lazy tensor with the upper bound x_lazy.size(0)
+  SymInt y0_lazy_size = y_lazy.sym_sizes()[0];
+  auto b_lazy = b_cpu.to(kLazy);
+  auto z_lazy = b_lazy.narrow_copy_symint(0, 0, y0_lazy_size);
+  ASSERT_EQ(y_lazy.sizes()[0], 10);
```
Comment (JackCaoG) [inline test/cpp/lazy/test_lazy_ops.cpp:7330]: is this checking the upper bound?

File: test/cpp/lazy/test_lazy_ops.cpp [hunk 2]
```diff
@@ -7311,6 +7311,26 @@ TEST_F(LazyOpsTest, TestNarrowCopy) {
   }
 }
 
+TEST_F(LazyOpsTest, NonZeroNarrowCopy) {
+  FLAGS_ltc_enable_symbolic_shapes = true;
+  auto x_cpu = torch::rand({10});
+  auto y_cpu = torch::nonzero(x_cpu);
+  int64_t y0_cpu_size = y_cpu.sizes()[0];
+  auto b_cpu =
```
Comment (wconstab) [inline test/cpp/lazy/test_lazy_ops.cpp:7319]: i'm confused why you have a separate x_cpu and b_cpu tensor.

also, is torch::rand a good way to exercise nonzero?  you'll get 10 nonzero elements some really high percent of the time

File: test/cpp/lazy/test_lazy_ops.cpp [hunk 3]
```diff
@@ -7311,6 +7311,26 @@ TEST_F(LazyOpsTest, TestNarrowCopy) {
...
+      torch::randn({10}); // a base tensor so we can compute with symint
+  auto z_cpu = b_cpu.narrow_copy(0, 0, y0_cpu_size);
+
+  // Same operations, but on LAZY tensor
+  auto x_lazy = x_cpu.to(kLazy);
+  auto y_lazy = torch::nonzero(x_lazy);
+  // y_lazy is a lazy tensor with the upper bound x_lazy.size(0)
+  SymInt y0_lazy_size = y_lazy.sym_sizes()[0];
+  auto b_lazy = b_cpu.to(kLazy);
+  auto z_lazy = b_lazy.narrow_copy_symint(0, 0, y0_lazy_size);
```
[the remark this answers is not in the document]
Comment (Krovatkin) [reply test/cpp/lazy/test_lazy_ops.cpp:7329]: This test comes from https://docs.google.com/document/d/1qUBwlLbDPsWKP3kaSCEgIF4hf-C4vvpA10upEvZep_s/edit# page 31. It's supposed to test there are no exceptions when we plumb a SymInt from nonzero through another op.

File: test/cpp/lazy/test_lazy_ops.cpp (hunk 1 above)
Comment (Krovatkin) [reply test/cpp/lazy/test_lazy_ops.cpp:7330]: correct, this is the test we included in https://docs.google.com/document/d/1qUBwlLbDPsWKP3kaSCEgIF4hf-C4vvpA10upEvZep_s/edit# p31

File: test/cpp/lazy/test_lazy_ops.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (wconstab) [reply test/cpp/lazy/test_lazy_ops.cpp:7329]: ah. ok, might want to just write a comment saying that, but its fine

**Pull request closed without merging**